### PR TITLE
Buffer JSON slog messages

### PR DESCRIPTION
**Pull request by hcstern**
serde_json will happily do a `write` syscall for every token it serializes. Add a small 100-character BufWriter buffer to prevent excessive syscalls.

Note that serde_json does not appear to call `Write::flush` so short lines will not be printed until the buffer is full.

### CODE REVIEW

**Comment by hcstern** [comment]
The non-flushing behavior is probably unexpected and might make debugging confusing but I don't think is a problem in prod. Would it make sense to only turn on the bufwriter behavior with a release profile?

**Comment by inikulin** [comment]
> The non-flushing behavior is probably unexpected and might make debugging confusing but I don't think is a problem in prod. Would it make sense to only turn on the bufwriter behavior with a release profile?

I think it's fine if we enable https://docs.rs/slog-json/latest/slog_json/struct.JsonBuilder.html#method.set_flush

**Comment by hcstern** [comment]
> > The non-flushing behavior is probably unexpected and might make debugging confusing but I don't think is a problem in prod. Would it make sense to only turn on the bufwriter behavior with a release profile?
> 
> I think it's fine if we enable https://docs.rs/slog-json/latest/slog_json/struct.JsonBuilder.html#method.set_flush

Nice catch, thanks! I'll increase the buffer size as well.

File: foundations/src/telemetry/log/init.rs [hunk 1]
```diff
@@ -70,6 +71,9 @@ pub(crate) fn init(service_info: &ServiceInfo, settings: &LoggingSettings) -> Bo
     // NOTE: OXY-178, default is 128 (https://docs.rs/slog-async/2.7.0/src/slog_async/lib.rs.html#251)
     const CHANNEL_SIZE: usize = 1024;
 
+    // buffer json log lines up to 500 characters
+    const BUF_SIZE: usize = 500;
```
Comment (bobrik) [inline foundations/src/telemetry/log/init.rs:75]: With lots of keys in json it's easy to exceed this. I'd go with something like 4k.

**Review by inikulin — APPROVED** [review]

File: foundations/src/telemetry/log/init.rs (hunk 1 above)
Comment (hcstern) [reply foundations/src/telemetry/log/init.rs]: Sorry, didn't see this last night and it didn't go to my cloudflare email for some reason even though I added it to this account. Thanks again!

**Pull request merged**